File: algo-palindromes/python/palindrome.py
def palindrome(argument):
    if type(argument) == int:
        return str(argument) == str(argument)[::-1]
    if len(argument) == 0:
        return True
    if type(argument) == str:
        if len(argument.split(" ")) == 1:
            return argument.lower() == argument.lower()[::-1]
        else:
            new_string = ""
            for letter in argument.lower():
                if letter.isalnum():
                    new_string += letter
            return new_string == new_string[::-1]

    return False

File: algo-palindromes/js/palindrome.js
const palindrome = function (word) {
  if (typeof word == "number" && word !== null) {
    const num = word.toString();
    let num_reverse = "";
    for (let i = num.length - 1; i >= 0; i--) {
      num_reverse += num[i];
    }
    return num == num_reverse;
  }
  if (word.length == 0) {
    return true;
  }
  let new_string = "";
  const alphanumeric = "abcdefghijklmnopqrstuvwxyz0123456789";
  for (let letter of word.toLowerCase()) {
    if (alphanumeric.indexOf(letter) !== -1) {
      new_string += letter;
    }
  }
  let reversed_string = "";
  for (let j = new_string.length - 1; j >= 0; j--) {
    reversed_string += new_string[j];
  }
  return reversed_string == new_string;
};

module.exports = { palindrome };
